ALEC-226: Update About chapter (#93)
OpenNMS/alec: docs/modules/about/pages/welcome.adoc

:imagesdir: ../assets/images

= About ALEC

The Architecture for Learning Enabled Correlation (ALEC) provides a machine learning-powered framework for logically grouping related alarms into higher-level situations.
As the correlation engines learn from users' network environments and their feedback, they will provide more accurate groupings and reduce the total number of notifications generated.

> Because correlated alarms are likely to have been generated by the same underlying issue, alarm groupings provide a starting point for root cause analysis.

Reviewer: ```suggestion
Because correlated alarms are likely to have been generated by the same underlying issue, alarm groupings provide a starting point for root-cause analysis.
```

== How does it work?

> First, your provisioned network inventory is analyzed and converted to a graphical representation.

Reviewer: Could we say what does the analysis?
```suggestion
First, X analyzes your provisioned network inventory and converts it to a graphical representation.
```

Author: Once this PR is merged, I'll cross-reference this sentence with the updated cluster.adoc page.

```suggestion
First, the cluster engine analyzes your provisioned network inventory and converts it to a graphical representation.
```

Inventory objects that are analyzed include nodes, their components, and their relations to other nodes.

.Example network inventory graph

> image::model.png[Network inventory graph that displays ten inventory objects and their directional relations, 600]

Reviewer: ```suggestion
image::model.png[Network inventory graph that displays ten inventory objects and their directional relations, 600]

```
We are missing a line space here, so it was running together with the next paragraph.

Author: Odd. That might be the build issues we've been running into, since there's already a line space between this and the next paragraph as usual.

After the graph is generated, alarms are enriched with additional metadata: the managed object type, and the alarm's managed object instance.
This metadata is required to identify which components the alarms are related to.

Alarms are then attached to the graph as they are triggered, and grouped into clusters based on their level of similarity and whether they share the same root cause.

> When ALEC determines that a cluster of alarms is related, an event is sent to OpenNMS.

Reviewer: ```suggestion
When ALEC determines that a cluster of alarms is related, it sends an event to OpenNMS.
```

OpenNMS then creates a new situation for the cluster.

.Example network inventory graph with alarm cluster and situation

> image::model_with_situation.png[Network inventory graph that displays ten inventory objects, their directional relations, a cluster of four attached alarms, and a situation, 600]

Reviewer: ```suggestion
image::model_with_situation.png[Network inventory graph that displays 10 inventory objects, their directional relations, a cluster of four attached alarms, and a situation, 600]

```

ALEC uses https://www.drools.org/[Drools] rules to manage the state of all alarms, including situations:

* The severity of a situation is the maximum severity of all related alarms, plus one.
* If all alarms related to a situation are acknowledged, the situation is also acknowledged.

== Correlation engines

ALEC implements two correlation engines:

* *Deep Learning:* an algorithm that leverages deep-learning techniques through a neural network to correlate and group alarms.

> * *DBSCAN:* leverages the https://en.wikipedia.org/wiki/DBSCAN[DBSCAN algorithm] to group alarms based on their distance from each other.

Author: ```suggestion
* *Clustering:* leverages the https://en.wikipedia.org/wiki/DBSCAN[DBSCAN algorithm] to group alarms based on their distance from each other.
```

> * *Deep Learning:* an algorithm that leverages deep learning techniques through a neural network to correlate and group alarms.

Reviewer: ```suggestion
* *Deep Learning:* an algorithm that leverages deep-learning techniques through a neural network to correlate and group alarms.
```

> To learn more, see xref:engines:deeplearning.adoc[Deep Learning] and xref:engines:dbscan.adoc[DBSCAN].

Author: ```suggestion
To learn more, see xref:engines:deeplearning.adoc[Deep Learning] and xref:engines:dbscan.adoc[Clustering].
```

> To learn more, see xref:engines:deeplearning.adoc[Deep Learning] and xref:engines:dbscan.adoc[DBSCAN].

Reviewer: These are not showing up as links, but that could just be a build thing.

Author: Could be... they show up as links in the VSC preview, so it should be okay. 🤔

:imagesdir: ../assets/images

= About ALEC

The Architecture for Learning Enabled Correlation (ALEC) provides a machine learning-powered framework for logically grouping related alarms into higher-level situations.

> As the correlation engines learn from users' network environments and their feedback, they will provide more accurate groupings and reduce the total number of notifications generated.

Reviewer 2: The current engines (DBSCAN and DeepLearning) are not using the feedback. It is intended for future iterations to use such feedback as we actively retrain the models.

Right now, the DBSCAN engine calculates the groupings at each run. As for the deep learning engine, my understanding is that it only gets trained once and then used. Might need to confirm this, but it has to get "manually" retrained with new/more data.

Author: ```suggestion
```
